dll/win32/kernel32/client/synch.c

```diff
@@ -988,4 +988,59 @@ UnregisterWaitEx(IN HANDLE WaitHandle,
     return TRUE;
 }
 
+#if (_WIN32_WINNT >= _WIN32_WINNT_VISTA) || (DLL_EXPORT_VERSION >= _WIN32_WINNT_VISTA)
```

Comment:
I really prefer the previous version, where all functions were in runonce.c instead of scattered into 2 different files in 2 different folders, with one function available to kernel32_vista, but the others not.

Reply:
`InitOnceExecuteOnce` exists in both of kernel32 and kernel32_vista indeed, but seems they should be.

The one exists in kernel32 will be complied and exported only if NT6+ build, the another one exists in kernel32_vista always exported for wine Dlls (see conversation above: Why isn't the InitOnceExecuteOnce function below inside this _WIN32_WINNT guard?).

If all functions in kernel32, kernel32 should always export `InitOnceExecuteOnce` for wine Dlls even if NT5 build, seems not right.

Reply:
kernel32 will only export these functions when `DLL_EXPORT_VERSION` is set to 0x600+. That is defined in the spec file. All the code in kernel32_vista is compiled into a static library (kernel32_vista_static) and linked to kernel32, so anything you put in there will be automatically included in kernel32, if you export it. There is no need to put the code directly into kernel32. There is also no need for guards, because these functions are tiny and we already export one of them from kernel32_vista anyway, so we can just compile and export all of them.